cli/azd/internal/vsrpc/server_session.go

```diff
@@ -0,0 +1,95 @@
+package vsrpc
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"sync"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/apphost"
+	"github.com/azure/azure-dev/cli/azd/pkg/ioc"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools/dotnet"
+	"go.lsp.dev/jsonrpc2"
+)
+
+type serverSession struct {
+	// container is the IoC container for the session. It is created during the InitializeAsync by scoping the
+	// root container from the server.
+	container *ioc.NestedContainer
+	// rootPath is the path to the root of the solution.
+	rootPath string
+	// sessionMu protects access to the session.
+	sessionMu sync.Mutex
+	// outWriter is the writer connected to stdout for all azd components in this session.
+	outWriter *writerMultiplexer
+	// outWriter is the writer connected to stderr for all azd components in this session.
+	errWriter *writerMultiplexer
+	// manifestCache is a cache of manifests for the session.
```

Comment:
> Is the key here the cleaned relative path to the manifest? Could be nice to document it to avoid programmer mishaps

You are right. The comment now reads:

```golang
// manifestCache is a cache of manifests for the session. The key is the full path to to the app host csproj file.
```